### ISSUE TITLE
Ability to search/show foods based on macros

### ISSUE THREAD

**Issue by 0brien** [davidhealey/waistline#692]
Sometimes I have, say, 500 calories left for the day, and I'm still 30g of protein short of my (minimum) goal. At these times it would be great if I could search my database for foods below a certain number of calories but above a certain amount of protein per serving. It would be even better if I could use a couple of these filters at once. For example: 'show me all foods that are below 500 calories, above 30g of protein, and above 10g of fat, per serving.'

I imagine that it might be hard to do this in the app. A next-best alternative would be to make this possible within the database export. This may already be possible - I couldn't find a way to convert the JSON into a format I could do this in, but then again I'm not very technically skilled. If it is possible, it would be great if someone could explain how.

Thanks for all the work you do, it's a great app, been using it for over a year!

**Comment by davidhealey** [comment]
>  imagine that it might be hard to do this in the app. 

I don't think it would be. I see a filter button that pops up a dialog where you can set the filter.
